cscs-checks/mch/alltoallv.py

```diff
@@ -0,0 +1,95 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.parameterized_test([''], ['--nocomm'], ['--nocomp'])
+class Alltoallv(rfm.RegressionTest):
+    def __init__(self, exec_parameter):
+        super().__init__()
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        if self.current_system.name in ['daint', 'dom']:
+            self.modules = ['craype-accel-nvidia60']
+            self._pgi_flags = '-acc -ta=tesla:cc60 -Mnorpath'
+        elif self.current_system.name in ['kesch']:
+            self.modules = ['craype-accel-nvidia35']
+            self._pgi_flags = '-O2 -ta=tesla,cc35,cuda8.0'
+
+        self.num_tasks = 144
+        self.num_gpus_per_node = 16
+        self.num_tasks_per_node = 16
+        self.num_tasks_per_socket = 8
+        self.executable = 'src/comm_overlap_benchmark %s' % exec_parameter
+        self.sourcesdir = ('https://github.com/cosunae/comm_overlap_bench')
+        self.sourcepath = 'src'
+        self.sanity_patterns = sn.assert_found(r'ELAPSED TIME:', self.stdout)
+        self.perf_patterns = {
+            'perf': sn.extractsingle(r'ELAPSED TIME:\s+(?P<perf>\S+)',
+                                     self.stdout, 'perf', float, 1)
+        }
+
+        if exec_parameter == '':
+            self.reference = {
+                'kesch:cn': {
+                    'perf': (5.53777, None, 0.15)
+                },
+            }
+        elif exec_parameter == '--nocomm':
+            self.reference = {
+                'kesch:cn': {
+                    'perf': (5.7878, None, 0.15)
+                },
+            }
+        elif exec_parameter == '--nocomp':
+            self.reference = {
+                'kesch:cn': {
+                    'perf': (5.62155, None, 0.15)
+                },
+            }
```

Comment:
There is too much code duplication in the above if/else. Rewrite it as follows:

```python
if not exec_parameter:
    ref = 5.53777
elif exec_parameter == '--nocomm':
    ref = 5.7878
elif exec_parameter == '--nocomp':
    ref = 5.62155

self.reference = {
    'kesch:cn': {
        'perf': (ref, None, 0.15)
    }
}
```

Reply:
done